### PR TITLE
Feat/asyncapi with victools

### PR DESCRIPTION
**Pull request by Fox32**
The PR replaces the implementation of the AsyncAPI generation based on [mbknor-jackson-jsonSchema](https://github.com/mbknor/mbknor-jackson-jsonSchema) with [victools](https://github.com/victools/jsonschema-generator).

Right now we have the following issues:
* the generated JSON schema is not nice as it contains definition names with counting suffixes.
* It is also desired to implement support for Jackson `defaultImpl`.
* Update documentation
* Contribute some parts back to victools

### CODE REVIEW

File: sda-commons-shared-asyncapi/README.md [hunk 1]
```diff
@@ -125,6 +125,8 @@ String expected = JsonSchemaGenerator
 
 ## Document Models
 
+// TODO: Update docs! mention new annotations and reference the library!
```
Comment (Fox32) [inline sda-commons-shared-asyncapi/README.md:128]: Documentation is missing here

File: sda-commons-shared-asyncapi/src/main/java/org/sdase/commons/shared/asyncapi/JsonSchemaGenerator.java [hunk 2]
```diff
@@ -77,17 +100,61 @@ public static SchemaBuilder builder() {
...
+      // TODO: This is behavior I would expect from the JacksonModule, should we contribute it?
+      configBuilder
+          .forFields()
+          .withRequiredCheck(
+              member -> {
+                JsonProperty jsonProperty =
+                    member.getAnnotationConsideringFieldAndGetter(JsonProperty.class);
+
+                return jsonProperty != null && jsonProperty.required();
+              });
```
Comment (Fox32) [inline sda-commons-shared-asyncapi/src/main/java/org/sdase/commons/shared/asyncapi/JsonSchemaGenerator.java:141]: We should consider contributing this to JacksonModule in victools

File: sda-commons-shared-asyncapi/src/main/java/org/sdase/commons/shared/asyncapi/schema/JsonSchemaExamplesModule.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.github.victools.jsonschema.generator.FieldScope;
+import com.github.victools.jsonschema.generator.Module;
+import com.github.victools.jsonschema.generator.SchemaGenerationContext;
+import com.github.victools.jsonschema.generator.SchemaGeneratorConfigBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO: Should we contribute this? This would be a good default behavior for victools.
```
Comment (Fox32) [inline sda-commons-shared-asyncapi/src/main/java/org/sdase/commons/shared/asyncapi/schema/JsonSchemaExamplesModule.java:14]: This file is something that we should contribute to victools. Having such a annotation there would be quite useful.

**Comment by JoergSiebahn** [comment]
Victools will also be here in the next major release, taken over from https://github.com/SDA-SE/sda-spring-boot-commons, see #2952.

Thanks @Fox32 for the bases long time ago. In the meantime also Victools improved a lot, though we are missing some features like [multiple examples](https://github.com/SDA-SE/sda-dropwizard-commons/pull/409/files#diff-ee20276853e859c22f6ba264e74fe7bcfc23e551d8cf2b81b26f969013653ea5). So far, we want to avoid infecting consumers with the custom annotation. Migrating from the old library and replacing uncommon annotations was already not the best journey.

**Pull request closed without merging**